rekognition_tutorial/short_sa_rekognition_tutorial.ipynb

```diff
@@ -0,0 +1,1222 @@
+   "id": "3b6d2788",
+   "metadata": {},
+   "source": [
+    "## Predict unlabeled images\n",
+    "<a id='predict_unlabeled_images'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5134cf3e",
+   "metadata": {},
+   "source": [
+    "We can now get the predictions for all our unlabeled images.\n",
+    "\n",
+    "We could use the function $\\textit{show_custom_labels}$ and dictionary with S3 paths to all unlabeld images $data\\_ links\\_dict$ that we created above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a9314fd6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predicted_labels  = {}\n",
+    "\n",
+    "for photo_path in tqdm(data_links_dict['unlabeled']):\n",
+    "    labels = show_custom_labels(rek_client, model_arn, bucket, photo_path, min_confidence)\n",
+    "    predicted_labels[photo_path] = labels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e5fc94a3",
+   "metadata": {},
+   "source": [
+    "## Make SA annotations\n",
+    "<a id='make_sa_annotations'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14f80c2f",
+   "metadata": {},
+   "source": [
+    "Based on predictions made by the model we should now create annotations in SuperAnnotate format to be able to upload them to SuperAnnotate."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9f92fb0d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ANNOTATIONS_FOLDER = 'PATH/TO/LOCAL/DIR/' # local folder to store .json files with annotations\n",
+    "for image_path, label in predicted_labels.items():\n",
+    "    filename = image_path.split('/')[-1]\n",
+    "    js_annotation = {\"metadata\": {\"name\": filename},\n",
+    "                     \"instances\": [{\"type\": \"tag\",\n",
+    "                                    \"className\": label[0]['Name']}]}\n",
+    "    with open(f'{ANNOTATIONS FOLDER}/{filename}.json','w') as f:\n",
+    "        json.dump(js_annotation, f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3eff4e3e",
+   "metadata": {},
+   "source": [
+    "## Upload new annotations to SA \n",
+    "<a id='upload_new_annotations_to_sa'></a> "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "01a3528e",
+   "metadata": {},
+   "source": [
+    "Now we could upload annotations generated on the previous step back to SuperAnnnotate."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "329bea86",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "outputs = []\n",
+    "full_dirname = ANNOTATIONS_FOLDER\n",
+    "files = os.listdir(full_dirname)\n",
+    "files_per_step = 500\n",
+    "steps = len(files) // files_per_step + 1\n",
+    "for step in range(steps):\n",
+    "    start = step * files_per_step\n",
+    "    end = min((step + 1)* files_per_step, len(files))\n",
+    "\n",
+    "    batch = [read(os.path.join(full_dirname, f)) for f in files[start: end]]\n",
```

Comment:
What is read, I can't find any import or function? Is it ok?

Reply:
+